### PR TITLE
[java] Add jersey3 support to the Java client

### PR DESCRIPTION
**Pull request by wing328**
- Add jersey3 support to the Java client

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### CODE REVIEW

File: samples/client/petstore/java/jersey3/build.gradle [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+}
+
+ext {
+    swagger_annotations_version = "1.6.5"
+    jackson_version = "2.13.2"
+    jackson_databind_version = "2.13.2"
+    jackson_databind_nullable_version = "0.2.2"
+    jakarta_annotation_version = "1.3.5"
+    jersey_version = "2.35"
```
Comment (andi-livn) [inline samples/client/petstore/java/jersey3/build.gradle:106]: Gradle still referring to Jersey 2.35

File: samples/client/petstore/java/jersey3/build.sbt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    version := "1.0.0",
+    scalaVersion := "2.11.4",
+    scalacOptions ++= Seq("-feature"),
+    Compile / javacOptions ++= Seq("-Xlint:deprecation"),
+    Compile / packageDoc / publishArtifact := false,
+    resolvers += Resolver.mavenLocal,
+    libraryDependencies ++= Seq(
+      "com.google.code.findbugs" % "jsr305" % "3.0.0",
+      "io.swagger" % "swagger-annotations" % "1.6.5",
+      "org.glassfish.jersey.core" % "jersey-client" % "2.35",
```
Comment (andi-livn) [inline samples/client/petstore/java/jersey3/build.sbt:14]: SBT still referring to Jersey 2.35

Comment (wing328) [reply samples/client/petstore/java/jersey3/build.sbt:14]: Good catch. I'll update it.

Comment (wing328) [reply samples/client/petstore/java/jersey3/build.sbt]: done via https://github.com/OpenAPITools/openapi-generator/pull/12046/commits/4f8519c2de74e04a056f7abd351a5a500159f971

File: samples/client/petstore/java/jersey3/build.gradle (hunk 1 above)
Comment (wing328) [reply samples/client/petstore/java/jersey3/build.gradle]: done via https://github.com/OpenAPITools/openapi-generator/pull/12046/commits/4f8519c2de74e04a056f7abd351a5a500159f971

**Review by andi-livn — COMMENTED** [review]
I know very little about mustache templating, but assuming this was based on the jersey2 template by replacing the necessary dependencies and client imports, this looks good. Thanks @wing328

**Pull request merged**